### PR TITLE
Swagger documentation for user authentication

### PR DESCRIPTION
**Pull request by ebleiweiss-r7**
Adds swagger api documentation for new user authentication functionality
MS-3294

## Verification

List the steps needed to make sure this thing works

- [x] Start `msfdb_ws`
- [x] Click the Try it out button on any endpoint, and Execute the function
- [x] **Verify** that you get a 401 Unauthorized Error
- [x] **Verify** that the new Swagger entry for `auth` displays near the top of the page
- [x] Click the Try it out button on the auth endpoint, enter your username and password, then click "Execute"
- [x] **Verify** that you get a 200 response that returns an API token
- [x] Click the Authorize button at the top-right of the page, and paste in the API token you just generated
- [x] **Verify** that any API endpoint is now authorized, and returns a 200 response
- [x] **Verify** that the new `user` GET and POST methods display along with their proper request/response models
- [x] **Verify** that all endpoints display the new 401 response model

### CODE REVIEW

File: documentation/api/v1/auth_api_doc.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    # Swagger documentation for /api/v1/auth/generate-token GET
+    operation :get do
+
+      key :description, 'Return a valid Authorization Bearer token.'
+      key :tags, [ 'auth' ]
+
+      parameter do
+        key :name, :username
+        key :in, :query
+        key :description, ''
```
Comment (jbarnett-r7) [inline documentation/api/v1/auth_api_doc.rb]: I know it's pretty self-explanatory, but can we add a basic description for these properties?

File: documentation/api/v1/credential_api_doc.rb [hunk 2]
```diff
@@ -167,6 +167,13 @@ module CredentialApiDoc
         end
       end
 
+      response 401 do
+        key :description, 'Authenticate to access this resource.'
```
Comment (jbarnett-r7) [inline documentation/api/v1/credential_api_doc.rb]: I know I didn't do it for the response 500 description, but we should change this to a constant that lives in `root_api_doc.rb` and reference it in each model. Can you please make the change for 401. And if you're feeling ambitious, 500, too?

Comment (ebleiweiss-r7) [reply documentation/api/v1/credential_api_doc.rb]: Are you talking about just description string? I couldn't find a way to make the 401s and 500s default responses without explicitly including them in every file. The string I can do, though!

File: documentation/api/v1/user_api_doc.rb [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            key :type, :array
+            items do
+              key :'$ref', :User
+            end
+          end
+        end
+      end
+
+      response 200 do
+        key :description, 'Authenticate to access this resource'
```
Comment (jbarnett-r7) [inline documentation/api/v1/user_api_doc.rb]: Copy/paste error I think?

File: documentation/api/v1/user_api_doc.rb [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      key :description, 'Create a user.'
+      key :tags, [ 'user' ]
+
+      parameter do
+        key :in, :body
+        key :name, :body
+        key :description, 'The attributes to assign to the user.'
+        key :required, true
+        schema do
+          property :username, type: :string, required: true, description: USERNAME_DESC
```
Comment (jbarnett-r7) [inline documentation/api/v1/user_api_doc.rb]: Can you add examples for these, please?

File: documentation/api/v1/credential_api_doc.rb (hunk 2 above)
Comment (jbarnett-r7) [reply documentation/api/v1/credential_api_doc.rb]: Yeah, just the description string. I actually spent way too much time trying to figure out how to make those responses default, but ultimately couldn't get it to work. This appears the be the best solution for now.

**Comment by busterb** [comment]
## Release Notes

This adds documentation for the Metasploit 5 user authentication API.

**Pull request merged**